contract/migrations/12.0.4.0.0/post-migration.py

```diff
@@ -0,0 +1,145 @@
+# Copyright 2019 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+
+from openupgradelib import openupgrade
+
+_logger = logging.getLogger(__name__)
+
+
+def migrate(cr, version):
+    cr.execute(
+        """
+        INSERT INTO contract_contract (
+            id,
+            name,
+            partner_id,
+            pricelist_id,
+            contract_type,
+            journal_id,
+            company_id,
+            analytic_account_id,
+            active,
+            code,
+            group_id,
+            contract_template_id,
+            recurring_invoices,
+            user_id,
+            recurring_next_date,
+            date_end,
+            payment_term_id,
+            fiscal_position_id,
+            invoice_partner_id,
+            message_main_attachment_id,
+            create_uid,
+            create_date,
+            write_uid,
+            write_date
+        )
+        SELECT id,
+               name,
+               partner_id,
+               pricelist_id,
+               contract_type,
+               journal_id,
+               company_id,
+               id,
+               active,
+               code,
+               group_id,
+               contract_template_id,
+               recurring_invoices,
+               user_id,
+               recurring_next_date,
+               date_end,
+               payment_term_id,
+               fiscal_position_id,
+               invoice_partner_id,
+               message_main_attachment_id,
+               create_uid,
+               create_date,
+               write_uid,
+               write_date
+        FROM account_analytic_account
+        WHERE recurring_invoices = TRUE
+        """
+    )
+    cr.execute(
+        """
+        INSERT INTO contract_line (
```

Comment:
No need to do this, but only renaming the table and the link field column (in pre).

Reply:
I couldn't do a renaming of the contract lines table in the pre-migration script because of the foreign key on contracts (which can't be renamed since analytic accounts must be preserved). So I have to migrate contracts and their lines the same way.

Reply:
@pedrobaeza,
As @ThomasBinsfeld mentioned, we can not rename the contract line table before we prepare the contract table.
How can we solve this problem?
I was looking for a way to create the contract table in the pre-migration script and insert the data there. Thus, the change of contract line name became possible.
I did not find an easy way to do it. In the end, it seems to me that the way things are done now is easy to understand if there is no decent alternative.